util/ng_corpus_generator.cpp

```diff
@@ -185,7 +185,7 @@ void findPaths(const NGHolder &g, CorpusProperties &cProps,
                 continue;
             }
 
-            if (!contains(one_way_in, v) &&
+            if (p && !contains(one_way_in, v) &&
```

Comment:
this is probably redundant, p is already used previously so definitely non-zero, you could probably just ignore this.